connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1053,16 +1064,29 @@ public void onRevoked(String leader, Collection<String> connectors, Collection<C
                 // this worker. Instead, we can let them continue to run but buffer any update requests (which should be
                 // rare anyway). This would avoid a steady stream of start/stop, which probably also includes lots of
                 // unnecessary repeated connections to the source/sink system.
-                for (String connectorName : connectors)
-                    worker.stopConnector(connectorName);
+                for (String connectorName : connectors) {
+                    if (worker.ownsConnector(connectorName)) {
+                        worker.stopConnector(connectorName);
```

Comment:
I think the other thing that's a bit confusing is how when you start a connector it needs try/catch, but stopping a connector doesn't (and handles its own failures). I get that this is because we have the extra `DistributedHerder.startConnector()` wrapper of `Worker.startConnector()`, but it feels asymmetric and confusing when reading this code.